yearn/contracts/CurveVoterProxyStrategy.sol

```diff
@@ -0,0 +1,511 @@
+
+        // Harvest the CRV rewards from the Curve pool's gauge.
+        IStrategyProxy(strategyProxy).harvest(tbtcCurvePoolGauge);
+
+        // Buy WBTC using harvested CRV tokens.
+        uint256 crvBalance = IERC20(crv).balanceOf(address(this));
+        if (crvBalance > 0) {
+            // Deposit a portion of CRV to the voter to gain CRV boost.
+            crvBalance = adjustCRV(crvBalance);
+
+            IERC20(crv).safeApprove(dex, 0);
+            IERC20(crv).safeApprove(dex, crvBalance);
+
+            address[] memory path = new address[](3);
+            path[0] = crv;
+            path[1] = weth;
+            path[2] = wbtc;
+
+            IUniswapV2Router(dex).swapExactTokensForTokens(
+                crvBalance,
+                uint256(0),
+                path,
+                address(this),
+                now
+            );
+        }
+
+        // Claim additional reward tokens from the gauge if applicable.
+        if (tbtcCurvePoolGaugeReward != address(0)) {
+            IStrategyProxy(strategyProxy).claimRewards(
+                tbtcCurvePoolGauge,
+                tbtcCurvePoolGaugeReward
+            );
+
+            uint256 rewardBalance = IERC20(tbtcCurvePoolGaugeReward).balanceOf(
+                address(this)
+            );
+            if (rewardBalance > 0) {
+                IERC20(tbtcCurvePoolGaugeReward).safeApprove(dex, 0);
+                IERC20(tbtcCurvePoolGaugeReward).safeApprove(
+                    dex,
+                    rewardBalance
+                );
+
+                address[] memory path = new address[](3);
+                path[0] = tbtcCurvePoolGaugeReward;
+                path[1] = weth;
+                path[2] = wbtc;
+
+                IUniswapV2Router(dex).swapExactTokensForTokens(
+                    rewardBalance,
+                    uint256(0),
+                    path,
+                    address(this),
+                    now
+                );
+            }
+        }
+
+        // Deposit acquired WBTC to the Curve pool to gain additional
+        // vault's underlying tokens.
+        uint256 wbtcBalance = IERC20(wbtc).balanceOf(address(this));
+        if (wbtcBalance > 0) {
+            IERC20(wbtc).safeApprove(tbtcCurvePoolDepositor, 0);
+            IERC20(wbtc).safeApprove(tbtcCurvePoolDepositor, wbtcBalance);
+            ICurvePool(tbtcCurvePoolDepositor).add_liquidity(
+                [0, 0, wbtcBalance, 0],
+                0
+            );
+        }
+
+        // Calculate the profit after obtaining new vault's underlying tokens.
+        profit = want.balanceOf(address(this)).sub(initialWantBalance);
+
+        // Check the profit and loss in the context of strategy debt.
+        uint256 totalAssets = estimatedTotalAssets();
+        uint256 totalDebt = vault.strategies(address(this)).totalDebt;
+        if (totalAssets < totalDebt) {
+            loss = totalDebt - totalAssets;
+            profit = 0;
```

Comment:
in this case we overwrite what was written on line 452. I guess it's no biggy, but maybe it will save us some gas if we do
```
if (totalAssets < totalDebt) {
//
profit = 0;
} else {
profit = want.balanceOf(address(this)).sub(initialWantBalance);
}

Reply:
Yep, makes sense. Please see https://github.com/keep-network/tbtc-v2/pull/8/commits/0790f4daead4c0b6cfbaee8c405f4815d76851a7